src/api/routes/v1/validator.ts

```diff
@@ -268,11 +268,11 @@ async function getChains(
   if (unl == null) {
     return undefined
   }
-  const results = await query('validators')
-    .select('chain')
+  const results = await query('networks')
+    .select('id')
     .distinct()
-    .where({ unl })
-  return results.map((result) => result.chain as string)
+    .where('unls', '~*', `(%,)?${unl}(,%)?`)
+  return results.map((result) => result.id as string)
```

Comment:
@mvadari This one is the change to the way we retrieve chain from unl to return validators for /v2/network/validators/{unl}. Previously the way we did it was to match any unl with chain in validators table but a lot of validators has unl of null now (in fact, all validators with chain amm-dev has unl of null). So I changed the way we retrieve chain from unl using network table and regex to avoid this situation. I feel like this is a better way of doing it.

Reply:
All the validators in that chain have a UNL of `null` because they didn't have a UNL. I don't understand why we should return validators with a UNL of `null` - that doesn't make sense, when we want all the chains that have the given UNL

Reply:
We don't currently really have support for networks without a UNL

Reply:
Currently, we still use validators with unl of null in prod:
https://data.xrpl.org/v1/network/validators/vl.ripple.com

Reply:
I'm confused as well

Reply:
It's a little easier to tell here: https://data.xrpl.org/v1/network/validators/vl.altnet.rippletest.net
The chain has at least one validator with the given UNL, and then all validators are pulled from those chains

Reply:
Oh I see. So the logic for /v2/network/validators/{unl} would be first find the chains that has at least one UNL that match the requested unl then pull all validators from those chains? If so, the old logic would be correct then, but /v2/network/validators/vlamm.devnet.rippletest.net will return an empty list since none of those validators has an UNL yet.

Reply:
Exactly, yeah

Reply:
One thing to note: you'll have to manually remove the data for that network from the production instance of the VHS first, since it was accidentally already added

Reply:
I reverted to old method. Not sure how to clean up on production yet, probably will reach out to you tmr for instructions.

Reply:
@ckniffen